gtda/homology/simplicial.py

```diff
@@ -720,3 +721,230 @@ def plot(Xt, sample=0, homology_dimensions=None):
         """
         return plot_diagram(
             Xt[sample], homology_dimensions=homology_dimensions)
+
+
+@adapt_fit_transform_docs
+class FlagserPersistence(BaseEstimator, TransformerMixin):
+    """:ref:`Persistence diagrams <persistence diagram>` resulting from
+    :ref:`filtrations <filtered complex>` of :ref:`directed or undirected flag
+    complexes <clique or flag complexes>`.
+
+    Given a weighted directed or undirected graph, information about the
+    appearance and disappearance of topological features (technically,
+    :ref:`homology classes <homology and cohomology>`) of various dimension
+    and at different scales is summarised in the corresponding persistence
+    diagram.
+
+    Parameters
+    ----------
+    homology_dimensions : list or tuple, optional, default: ``(0, 1)``
+        Dimensions (non-negative integers) of the topological features to be
+        detected.
+
+    directed : bool, optional, default: ``True``
+        If ``True``, :meth:`transform` computes the persistence diagrams of
+        filtered directed flag complexes arising from weighted directed
+        graphs. Otherwise, it computes the persistence diagrams of filtered
+        flag complexes arising from weighted undirected graphs. Hence, in
+        the latter case :meth:`transform` computes the same quantity as the
+        ``transform`` method of a :class:`VietorisRipsPersistence` object
+        instantiated with ``metric=precomputed``.
+
+    coeff : int prime, optional, default: ``2``
+        Compute homology with coefficients in the prime field
+        :math:`\\mathbb{F}_p = \\{ 0, \\ldots, p - 1 \\}` where
+        :math:`p` equals `coeff`.
+
+    max_entries : int, optional, default: ``-1``
+        Number controlling the degree of precision in the matrix
+        reductions performed by the the backend. Corresponds to the parameter
+        ``approximation`` in :func:`pyflagser.flagser`. Increase for higher
+        precision, decrease for faster computation. A good value is often
+        ``100000`` in hard problems.  A negative value computes highest
+        possible precision.
+
+    max_edge_length : float, optional, default: ``numpy.inf``
+        Upper bound on the maximum value of the filtration parameter. Points
+        whose distance is greater than this value will never be connected by
+        an edge, and topological features at scales larger than this value
+        will not be detected.
```

Comment:
This should be what happens. It should be the same behaviour as `VietorisRipsPersistence`.

Reply:
@wreise could you explain the origin of the doubt as to whether this is really what happens?

Reply:
Ok, nothing. I forgot that we apply post-processing where we replace `np.inf`, which by default is `max_edge_length`

Reply:
@wreise actually I now think you are right, having had some time to look into it better. It *looks* like this is not achieving quite the same as in `VietorisRipsPersistence`. In particular, before postprocessing is applied I do not think one will have infinite bars at all, and in particular no infinities which can be replaced with `infinity_values`.

Reply:
🤦‍♂ i check only for `ViertorisRipsPersistence` in the end, where it is ok.
Yes, we get unexpected results when, for example, `max_edge_length` is specified and `infinity_values=np.inf`are both passed.
To be more specific, in
```
Xs = np.array([[[0, 1, 10, 1],
                [1, 0, 1, 10],
                [10, 1, 0, 1],
                [1, 10, 1, 0]]]).astype(float)

FlagserPersistence(max_edge_length=3, infinity_values=np.inf).fit_transform(Xs)
```
generators in H_1 die at 3, instead of `np.inf`.